src/upgrade/assessmentManager.ts

```diff
@@ -175,30 +178,181 @@ async function getWorkspaceIssues(workspaceFolderUri: string): Promise<UpgradeIs
     return workspaceIssues;
 }
 
-async function getAllDependencies(projectNode: INodeData): Promise<PackageDescription[]> {
+const MAVEN_CONTAINER_PATH = "org.eclipse.m2e.MAVEN2_CLASSPATH_CONTAINER";
+const GRADLE_CONTAINER_PATH = "org.eclipse.buildship.core.gradleclasspathcontainer";
+
+/**
+ * Parse direct dependencies from pom.xml file.
+ * Also checks parent pom.xml for multi-module projects.
+ */
+function parseDirectDependenciesFromPom(pomPath: string): Set<string> {
+    const directDeps = new Set<string>();
+    try {
+        const pomContent = fs.readFileSync(pomPath, 'utf-8');
+        
+        // Extract dependencies from <dependencies> section (not inside <dependencyManagement>)
+        // First, remove dependencyManagement sections to avoid including managed deps
+        const withoutDepMgmt = pomContent.replace(/<dependencyManagement>[\s\S]*?<\/dependencyManagement>/g, '');
+        
+        // Match <dependency> blocks and extract groupId and artifactId
+        const dependencyRegex = /<dependency>\s*<groupId>([^<]+)<\/groupId>\s*<artifactId>([^<]+)<\/artifactId>/g;
+        let match;
+        while ((match = dependencyRegex.exec(withoutDepMgmt)) !== null) {
+            const groupId = match[1].trim();
+            const artifactId = match[2].trim();
+            // Skip property references like ${project.groupId}
+            if (!groupId.includes('${') && !artifactId.includes('${')) {
+                directDeps.add(`${groupId}:${artifactId}`);
+            }
+        }
+
+        // Check for parent pom in multi-module projects
+        const parentPomPath = path.join(path.dirname(pomPath), '..', 'pom.xml');
```

Comment:
this may not be the fact. I think you can find all the pom.xml in this project first and then parse them one by one to find all the dependencies.